### PR TITLE
Bugfix for checking SyslogLogIDSEvents in syslog.LogTypes field

### PR DESCRIPTION
**Pull request by stevegaossou**
## Description
- Fixes bug in check for SyslogLogIDSEvents value in syslog.LogTypes field for managed cluster scenario.

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: pkg/controller/logcollector/logcollector_controller.go
Comment (stevegaossou) [inline pkg/controller/logcollector/logcollector_controller.go:285]: This was the bug.

File: pkg/controller/logcollector/logcollector_controller.go
Comment (stevegaossou) [inline pkg/controller/logcollector/logcollector_controller.go:261]: No real reason to name the error mccErr, so switching it back to standard `err` name we always use.

Comment (matthewdupre) [reply pkg/controller/logcollector/logcollector_controller.go:261]: Either this change is wrong, or the existing code was.  Note that in line 265 we look at err (before), so this is an actual behavioral change.

**Review by caseydavenport — APPROVED** [review]

File: pkg/controller/logcollector/logcollector_controller.go
Comment (caseydavenport) [reply pkg/controller/logcollector/logcollector_controller.go:261]: I believe that the old code was wrong, based on my reading of it.

Comment (matthewdupre) [reply pkg/controller/logcollector/logcollector_controller.go:261]: This changes what we return in line 287.  I think that might be OK, but it's not just cosmetic?

Comment (stevegaossou) [reply pkg/controller/logcollector/logcollector_controller.go:261]: This ultimately did not cause the bug that was caught. However it did not match with the remaining references to the error in this section (which were using `err`). So either way the name had to get reconciled, either change all the references to `mccErr` or to `err`. 

I chose the latter.

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and delete the branch after I've merged it.

**Comment by marvin-tigera** [comment]
I'm sorry but I failed delete the branch after merging the pull request.

**Pull request merged**
